Panache @BsonIgnore for methods

## Describe the bug
Since Panache is more Domain Driven, I added also some methods with small logic which like convert a different input parameter e.g. Date to LocalDateTime or get max Date from an Array... 
But for some reason `@BsonIgnore` only works on additional methods which are not setting fields.

### Expected behaviour
I expect that the method which I mark with `@BsonIgnore` gets actually ignored

### Actual behaviour
Example of a method which updates  my `lastActivity` (`LocalDateTime`) by the max parameter of an `LocalDateTime[]` (Array)
```
//Additional Method
public User setLastActivity(final LocalDateTime... lastActivities) {
        this.lastActivity = Arrays.stream(lastActivities).max(LocalDateTime::compareTo).orElse(getTimeLastYear());
        return this;
}

//Normal Method
public User setLastActivity(LocalDateTime lastActivity) {
        this.lastActivity = lastActivity;
        return this;
}
```
Error on save or get:
 
```
org.bson.codecs.configuration.CodecConfigurationException: Property 'lastActivity' in User, has differing data types: TypeData{type=LocalDateTime[]} and TypeData{type=LocalDateTime}.
```
Error on save or get when renaming the method:
```
Unable to get value for property 'lastActivities' in User

A custom Codec or PojoCodec may need to be explicitly configured and registered to handle this type.
```

## To Reproduce
* Create a method to update manually a field on the object with different input parameter - in my example an array
* Add `@BsonIgnore` to it
* Error is happening when you try to save or get the object

@loicmathieu its the same class as above... In that example I have the field `lastActivity` and want to have a custom setter with an array. As described at the start it doesn't matter how I name my additional method as I always get strange errors like `Unable to get value for property`. I tested it now with void instead of returning the object itself without any difference. Its somehow impossible to add additional methods which is annoying as I need now always wrapper class with back and forward translation of the object. So this nice domain driven idea of panache from my side useless as I can't enhance my methods...
Also when I am able to add `@BsonIgnore` to a method then I would expect that this being recognised.

Method:
```java
    public void setLastActivities2(final LocalDateTime... lastActivities) {
        setLastActivity(Arrays.stream(lastActivities).max(LocalDateTime::compareTo).orElse(getTimeLastYear()));
    }
```
Error:
```
Unable to get value for property 'lastActivities2' in User

A custom Codec or PojoCodec may need to be explicitly configured and registered to handle this type.

	at org.bson.codecs.pojo.AutomaticPojoCodec.encode(AutomaticPojoCodec.java:53)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.getBsonDocument(MongoOperations.java:279)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.persistOrUpdate(MongoOperations.java:258)
	at io.quarkus.mongodb.panache.runtime.MongoOperations.persistOrUpdate(MongoOperations.java:162)
	at io.quarkus.mongodb.panache.PanacheMongoEntityBase.persistOrUpdate(PanacheMongoEntityBase.java:848)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:939)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:845)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: org.bson.codecs.configuration.CodecConfigurationException: Unable to get value for property 'lastActivities2' in User
	at org.bson.codecs.pojo.PropertyAccessorImpl.getError(PropertyAccessorImpl.java:69)
	at org.bson.codecs.pojo.PropertyAccessorImpl.get(PropertyAccessorImpl.java:45)
	at org.bson.codecs.pojo.PojoCodecImpl.encodeProperty(PojoCodecImpl.java:168)
	at org.bson.codecs.pojo.PojoCodecImpl.encode(PojoCodecImpl.java:106)
	at org.bson.codecs.pojo.AutomaticPojoCodec.encode(AutomaticPojoCodec.java:50)
	... 76 more
Caused by: org.bson.codecs.configuration.CodecConfigurationException: Unable to get value for property 'lastActivities2' in User
	at org.bson.codecs.pojo.PropertyAccessorImpl.getError(PropertyAccessorImpl.java:69)
	at org.bson.codecs.pojo.PropertyAccessorImpl.get(PropertyAccessorImpl.java:42)
	... 79 more
```

--- a/docs/src/main/asciidoc/mongodb-panache.adoc
+++ b/docs/src/main/asciidoc/mongodb-panache.adoc
@@ -7,7 +7,7 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 
 include::./attributes.adoc[]
 :config-file: application.properties
-:mongodb-doc-root-url: https://mongodb.github.io/mongo-java-driver/3.11
+:mongodb-doc-root-url: https://mongodb.github.io/mongo-java-driver/4.2
 
 MongoDB is a well known NoSQL Database that is widely used, but using its raw API can be cumbersome as you need to express your entities and your queries as a MongoDB link:{mongodb-doc-root-url}/bson/documents/#document[`Document`].
 
@@ -138,7 +138,7 @@ public class Person extends PanacheMongoEntity {
 
 NOTE: Annotating with `@MongoEntity` is optional. Here the entity will be stored in the `ThePerson` collection instead of the default `Person` collection.
 
-MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos/[PojoCodecProvider] to map your entities to a MongoDB `Document`.
+MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos/[PojoCodecProvider] to convert your entities to a MongoDB `Document`.
 
 You will be allowed to use the following annotations to customize this mapping:
 
@@ -289,7 +289,7 @@ public class Person  {
 
 NOTE: Annotating with `@MongoEntity` is optional. Here the entity will be stored in the `ThePerson` collection instead of the default `Person` collection.
 
-MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos/[PojoCodecProvider] to map your entities to a MongoDB `Document`.
+MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos/[PojoCodecProvider] to convert your entities to a MongoDB `Document`.
 
 You will be allowed to use the following annotations to customize this mapping:
 
@@ -518,7 +518,7 @@ using the same field multiple times in a query is not allowed using PanacheQL as
 (see link:https://github.com/quarkusio/quarkus/issues/12086[this issue on github]).
 
 We also handle some basic date type transformations: all fields of type `Date`, `LocalDate`, `LocalDateTime` or `Instant` will be mapped to the
-link:https://docs.mongodb.com/manual/reference/bson-types/#document-bson-type-date[BSON Date] using the `ISODate` type (UTC datetime).
+link:https://docs.mongodb.com/manual/reference/bson-types/#date[BSON Date] using the `ISODate` type (UTC datetime).
 The MongoDB POJO codec doesn't support `ZonedDateTime` and `OffsetDateTime` so you should convert them prior usage.
 
 MongoDB with Panache also supports extended MongoDB queries by providing a `Document` query, this is supported by the find/list/stream/count/delete methods.
@@ -654,6 +654,21 @@ This can be achieved by setting to DEBUG the following log category inside your 
 quarkus.log.category."io.quarkus.mongodb.panache.runtime".level=DEBUG
 ----
 
+== The PojoCodecProvider: easy object to BSON document conversion.
+
+MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos[PojoCodecProvider], with link:{mongodb-doc-root-url}/pojos/#pojo-support[automatic POJO support,
+to automatically convert your object to a BSON document.
+
+In case you encounter the `org.bson.codecs.configuration.CodecConfigurationException` exception, it means the codec is not able to
+automatically convert your object.
+This codec obeys the Java Bean standard, so it will successfully convert a POJO using public fields or getters/setters.
+You can use `@BsonIgnore` to make a field, or a getter/setter, ignored by the codec.
+
+If your class doesn't obey these rules (for example by including a method that starts with `get` but is not a setter),
+you could provide a custom codec for it.
+Your custom codec will be automatically discovered and registered inside the codec registry.
+See link:mongodb#simplifying-mongodb-client-usage-using-bson-codec[Using BSON codec].
+
 == Transactions
 
 MongoDB offers ACID transactions since version 4.0.
@@ -724,7 +739,7 @@ Kotlin data classes are a very convenient way of defining data carrier classes, 
 But this type of class comes with some limitations: all fields needs to be initialized at construction time or be marked as nullable,
 and the generated constructor needs to have as parameters all the fields of the data class.
 
-MongoDB with Panache uses the `PojoCodec`, a MongoDB codec which mandates the presence of a parameterless constructor.
+MongoDB with Panache uses the link:{mongodb-doc-root-url}/bson/pojos[PojoCodecProvider], a MongoDB codec which mandates the presence of a parameterless constructor.
 
 Therefore, if you want to use a data class as an entity class, you need a way to make Kotlin generate an empty constructor.
 To do so, you need to provide default values for all the fields of your classes.
@@ -734,10 +749,10 @@ __On the JVM, if the generated class needs to have a parameterless constructor, 
 
 If for whatever reason, the aforementioned solution is deemed unacceptable, there are alternatives.
 
-First, you can create a BSON Codec which will be automatically registered by Quarkus and will be used instead of the `PojoCodec`.
+First, you can create a BSON Codec which will be automatically registered by Quarkus and will be used instead of the `PojoCodecProvider`.
 See this part of the documentation: link:mongodb#simplifying-mongodb-client-usage-using-bson-codec[Using BSON codec].
 
-Another option is to use the `@BsonCreator` annotation to tell the `PojoCodec` to use the Kotlin data class default constructor,
+Another option is to use the `@BsonCreator` annotation to tell the `PojoCodecProvider` to use the Kotlin data class default constructor,
 in this case all constructor parameters have to be annotated with `@BsonProperty`: see link:{mongodb-doc-root-url}/bson/pojos/#supporting-pojos-without-no-args-constructors[Supporting pojos without no args constructor].
 
 This will only work when the entity extends `PanacheMongoEntityBase` and not `PanacheMongoEntity`, as the ID field also needs to be included in the constructor.

